soc/sifive: differentiate between FE310-G000 and FE310-G002

fkokosinski opened the pull request:
This PR reflects a difference between FE310-G000 and FE310-G002 SoCs, since only the latter supports PMP. The result of that is the split of the HiFive1 board into two separate targets, since the HWMv2 right now assumes that board revisions share the same SoC.

Fixes #90094.

fkokosinski committed:
soc/sifive: differentiate between FE310-G000 and FE310-G002

This commit reflects a difference between FE310-G000 and FE310-G002 SoCs,
since only the latter supports PMP. The result of that is the split of the
HiFive1 board into two separate targets, since the HWMv2 right now assumes
that board revisions share the same SoC.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

fkokosinski commented:
@adamatom - could you give this a spin?

fkokosinski commented:
The only test failing is `portability.xsi.single_process.newlib`, where `putenv` sets `errno` to `ENOMEM`, which sounds like the filters set for this tests are not string enough when it comes to memory size checking. I guess with newlib taking a big chunk out of the memory, there's no space really for the test of the test.

cc @cfriedt since this is in `tests/posix/`

adamatom commented:
> @adamatom - could you give this a spin?

My rev A boots with these changes:
```
*** Booting Zephyr OS build v1.7.99-102737-ga75780b2fc6c ***
```

and my `build/zephyr/include/generated/zephyr/autoconf.h` does not define `CONFIG_RISCV_PMP`, so it looks good from my point of view. Thanks for the fast turn-around.

adamatom commented:
> Zephyr OS build v1.7.99

Just noticed this dropped from 4.1.0 :sweat_smile: - the git hash matches your commit though: `a75780b2fc6c`.

ycsin commented:
> The only test failing is `portability.xsi.single_process.newlib`

[tests/kernel/msgq/msgq_usage/kernel.message_queue.usage](https://github.com/zephyrproject-rtos/zephyr/actions/runs/15138677546/job/42557072258?pr=90217) also failed:

```
13:55:19.2563 [INFO] uart0: [host: 0.14s (+0.14s)|virt: 36.1µs (+36.1µs)] ASSERTION FAIL [heap_sz > chunksz(sizeof(struct z_heap))] @ WEST_TOPDIR/zephyr/lib/heap/heap.c:549
13:55:19.2563 [INFO] uart0: [host: 0.14s (+0.21ms)|virt:     36.1µs (+0s)] heap size is too small
13:55:19.2565 [INFO] uart0: [host: 0.14s (+0.16ms)|virt:     36.1µs (+0s)] E: 
13:55:19.2567 [INFO] uart0: [host: 0.14s (+0.24ms)|virt:     36.1µs (+0s)] E:  mcause: 11, Environment call from M-mode
13:55:19.2569 [INFO] uart0: [host: 0.14s (+0.13ms)|virt:     36.1µs (+0s)] E:   mtval: 0
13:55:19.2571 [INFO] uart0: [host: 0.14s (+0.21ms)|virt:     36.1µs (+0s)] E:      a0: 00000004    t0: 00000000
13:55:19.2572 [INFO] uart0: [host: 0.14s (+0.16ms)|virt:     36.1µs (+0s)] E:      a1: 00000225    t1: 00000008
13:55:19.2574 [INFO] uart0: [host:  0.14s (+0.2ms)|virt:   80µs (+43.9µs)] E:      a2: 80003e84    t2: 00000000
13:55:19.2576 [INFO] uart0: [host: 0.14s (+0.18ms)|virt:       80µs (+0s)] E:      a3: 0000000a    t3: 00000010
13:55:19.2578 [INFO] uart0: [host: 0.14s (+0.17ms)|virt:       80µs (+0s)] E:      a4: 00000000    t4: 00000005
13:55:19.2585 [INFO] uart0: [host: 0.14s (+0.71ms)|virt:    0.1ms (+20µs)] E:      a5: 10013000    t5: 00000003
13:55:19.2587 [INFO] uart0: [host: 0.14s (+0.19ms)|virt:      0.1ms (+0s)] E:      a6: 00000000    t6: 00000000
13:55:19.2588 [INFO] uart0: [host: 0.14s (+0.11ms)|virt:      0.1ms (+0s)] E:      a7: 0000002d
13:55:19.2589 [INFO] uart0: [host: 0.14s (+0.12ms)|virt:      0.1ms (+0s)] E:      sp: 80003ea0
13:55:19.2590 [INFO] uart0: [host: 0.14s (+0.11ms)|virt:      0.1ms (+0s)] E:      ra: 204009c6
13:55:19.2591 [INFO] uart0: [host: 0.14s (+0.11ms)|virt:      0.1ms (+0s)] E:    mepc: 20400b92
13:55:19.2592 [INFO] uart0: [host: 0.14s (+0.11ms)|virt:      0.1ms (+0s)] E: mstatus: 00001880
13:55:19.2593 [INFO] uart0: [host:   0.14s (+37µs)|virt:      0.1ms (+0s)] E: 
13:55:19.2594 [INFO] uart0: [host: 0.14s (+0.18ms)|virt:      0.1ms (+0s)] E:      s0: 20406000    s6: 00000000
13:55:19.2596 [INFO] uart0: [host: 0.14s (+0.18ms)|virt:      0.1ms (+0s)] E:      s1: 20405da8    s7: 00000000
13:55:19.2598 [INFO] uart0: [host: 0.14s (+0.18ms)|virt:      0.1ms (+0s)] E:      s2: 80003ff0    s8: 00000000
13:55:19.2600 [INFO] uart0: [host: 0.14s (+0.18ms)|virt:      0.1ms (+0s)] E:      s3: 00000001    s9: 00000000
13:55:19.2602 [INFO] uart0: [host: 0.14s (+0.18ms)|virt:      0.1ms (+0s)] E:      s4: 00000000   s10: 00000000
13:55:19.2603 [INFO] uart0: [host: 0.14s (+0.17ms)|virt:      0.1ms (+0s)] E:      s5: 00000000   s11: 00000000
13:55:19.2604 [INFO] uart0: [host:   0.14s (+35µs)|virt:      0.1ms (+0s)] E: 
13:55:19.2605 [INFO] uart0: [host:   0.14s (+91µs)|virt:      0.1ms (+0s)] E: call trace:
13:55:19.2608 [INFO] uart0: [host: 0.14s (+0.31ms)|virt:      0.1ms (+0s)] E:       0: sp: 80003ea0 ra: 20400b92
13:55:19.2610 [INFO] uart0: [host: 0.14s (+0.21ms)|virt:      0.1ms (+0s)] E:       1: sp: 80003ec0 ra: 20401118
13:55:19.2612 [INFO] uart0: [host: 0.14s (+0.22ms)|virt:   0.2ms (+0.1ms)] E:       2: sp: 80003ed0 ra: 204028a0
13:55:19.2614 [INFO] uart0: [host: 0.14s (+0.18ms)|virt:      0.2ms (+0s)] E:       3: sp: 80003ed8 ra: 204028a2
13:55:19.2616 [INFO] uart0: [host: 0.14s (+0.18ms)|virt:      0.2ms (+0s)] E:       4: sp: 80003ee0 ra: 204028bc
13:55:19.2617 [INFO] uart0: [host: 0.14s (+0.18ms)|virt:      0.2ms (+0s)] E:       5: sp: 80003f08 ra: 204028a2
13:55:19.2619 [INFO] uart0: [host: 0.14s (+0.19ms)|virt:      0.2ms (+0s)] E:       6: sp: 80003f10 ra: 20400b82
13:55:19.2620 [INFO] uart0: [host:   0.14s (+56µs)|virt:      0.2ms (+0s)] E: 
13:55:19.2623 [INFO] uart0: [host: 0.14s (+0.27ms)|virt:      0.2ms (+0s)] E: >>> ZEPHYR FATAL ERROR 4: Kernel panic on CPU 0
13:55:19.2625 [INFO] uart0: [host: 0.15s (+0.25ms)|virt:      0.2ms (+0s)] E: Current thread: 0x800004d0 (main)
13:55:19.2626 [INFO] uart0: [host: 0.15s (+0.11ms)|virt:      0.2ms (+0s)] E: Halting system

INFO    - /__w/zephyr/zephyr/twister-out/hifive1_fe310_g000/zephyr/tests/kernel/msgq/msgq_usage/kernel.message_queue.usage/handler.log
INFO    - 4 Iteration:
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue

INFO    - 941/941 hifive1/fe310_g000        kernel.message_queue.usage                         FAILED Timeout (renode 120.614s <zephyr>)
```

ycsin commented on boards/sifive/hifive1/Kconfig.defconfig:
```diff
@@ -1,12 +1,17 @@
+
+if BOARD_HIFIVE1_REVB
+
+config SYS_CLOCK_TICKS_PER_SEC
+	default 128
```
I think this can be factored out?

fkokosinski replied:
I've decided to just leave it at the Kconfig default (i.e. remove this from defconfig), since it's not needed and caused problems in `timespec_util` tests

fkokosinski committed:
tests: add `min_ram` for heap-using testcases

Two testcases have an implicit minimum memory requirement due to heap usage
and are failing with `ENOMEM` for some platforms:
* portability.xsi.single_process.newlib
* kernel.message_queue.usage

This commit filters out platform that can't satisfy this requirement.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>
